sched/scheduler/task_runner.go

```diff
@@ -138,6 +140,28 @@ func (r *taskRunner) run() error {
 			"tag":        taskErr.st.Tag,
 			"err":        taskErr,
 		}).Info("End task")
+
+	// Check if failure was due to failure to git clean. If so, kill worker
```

Comment:
The other problem I have with the scheduler approach is that it's clunky to tune this to act on consecutive failures. We aren't certain we don't have flaky one-off command failures, so it's reasonable in the case we solving for now to require a few failures in a row before terminating.